formulae/tests/test_design_matrices.py

```diff
@@ -1101,13 +1101,19 @@ def test_data_is_empty(data):
 
 
 def test_wrong_na_action(data):
-    with pytest.raises(ValueError, match="'na_action' must be either 'drop' or 'error'"):
+    with pytest.raises(ValueError, match="'na_action' must be either 'drop', 'error' or 'pass'"):
         design_matrices("y ~ x1", data, na_action="whatever")
 
 
 def test_drop_na_rows(data):
     data["x3"] = data["x3"].replace({3: np.nan})
-    design_matrices("y ~ x3", data, na_action="drop")
+    dm = design_matrices("y ~ x3", data, na_action="drop")
+    assert dm.common.as_dataframe().shape[0] == 15
+
+
+def test_pass_na_rows(data):
+    dm = design_matrices("y ~ x3", data, na_action="pass")
+    assert dm.common.as_dataframe().shape[0] == 20
```

Comment:
I'm not sure if this `"x3"` in `data` contains the missing values added in `test_drop_na_rows()`. I think we could repeat line 1109 here to make sure we have the same missing.

Or maybe I'm missing something?